File: Wirtualny_Swiat_Java/Projekt_PO_Java/src/model/Organisms/Plant.java
package model.Organisms;

import model.Organism;
import model.Point;
import model.World;

public abstract class Plant extends Organism {
    private final int SPREAD_CHANCE = 20;


    public Plant(World world, Point position, int power){
        super(world, position, power, 0);
    }


    @Override
    public void Action(){
        increaseAge();

        if (trySpread()){
            Point newPos = findPosition();
            if (newPos.getX() != getPosition().getX() && newPos.getY() != getPosition().getY()){
                Organism newOrganism = makeNewOrganism(newPos);
                world.addOrganism(newOrganism, newPos);
                world.addEvent(newOrganism.getNazwa() + " rozmnożył sie na pozycji " + newPos.getX() + ", " + newPos.getY() + ")");
            }
        }
    }

    private boolean trySpread(){
        int rand = random.nextInt(100);
        return rand < SPREAD_CHANCE;
    }

    @Override
    public void Collision(Organism attacker){
        if (attacker.getPower() >= getPower()){
            this.setAlive(false);
            world.addEvent(attacker.getNazwa() + " zabil " + getNazwa() + " na pozycji (" + getPosition().getX() + ", " + getPosition().getY() + ")");
        }
    }
}

File: WirtualnySwiatPython/plant.py
import random

from organism import Organism
from config import SPREAD_CHANCE

class Plant(Organism):
    def __init__(self, world, position, power):
        super().__init__(world, position, power, 0)

    def trySpread(self):
        ranNum = random.randint(0, 99)
        return ranNum < SPREAD_CHANCE

    def Action(self):
        self.increaseAge()
        if (self.trySpread()):
            newPos = self.findPosition()
            if newPos is not None:
                newOrganism = self.makeNewOrganism(newPos)
                self.world.organisms.append(newOrganism)
                self.world.addEvent(f"{newOrganism.getNazwa()} rozmnozyl sie na ({newPos.GetX()}, {newPos.GetY()})")

    def Collision(self, attacker):
        self.collisionHelper(attacker, self.getPosition())